src/Schema/Blueprint.php

```diff
@@ -42,9 +44,19 @@ class Blueprint extends SchemaBlueprint
     /**
      * Create a new schema blueprint.
      */
-    public function __construct(Connection $connection, string $collection)
+    public function __construct(Connection $connection, string $collection, ?Closure $callback = null)
     {
-        parent::__construct($collection);
+        // Parent constructor signature was changed in Laravel 12
+        // https://github.com/laravel/framework/commit/f29df4740d724f1c36385c9989569e3feb9422df#diff-68f714a9f1b751481b993414d3f1300ad55bcef12084ec0eb8f47f350033c24bR107
+        self::$hasConnectionInConstructor ??= (new ReflectionMethod(parent::class, '__construct'))->getParameters()[0]->getName() === 'connection';
```

Comment:
Is there a reason you didn't try this approach for the construction of Query\Grammar and Schema\Grammar objects in Connection.php?

Would it not have worked to avoid the PHPStan issue?

Reply:
For the construction of the Grammar classes, PHP allows passing extra arguments that are not used by the contructor/method. I used this approach to not rely too much on reflection at runtime.